File: shared/types/model_record.py
from dataclasses import dataclass, field

@dataclass(slots=True)
class Tag:
  id: int
  name: str

@dataclass(slots=True)
class Prompts:
  positive: str
  negative: str

@dataclass(slots=True)
class OldMetaJson :
  description: str
  community_links: str
  images: list[str]
  prompts: Prompts

@dataclass(slots=True)
class MetaJson :
  description: str
  community_links: str
  images_count: int
  prompts: Prompts

@dataclass(slots=True)
class ModelRecord:
  sha256: str
  path: str
  name: str
  type: str
  size_bytes: int
  created_at: int
  meta_json: MetaJson
  tags: list[Tag] = field(default_factory=list)

@dataclass(slots=True)
class PendingModelRecord:
  id: int
  path: str
  sha256: str
  name: str
  type: str
  size_bytes: int
  created_at: int
  meta_json: OldMetaJson
  tags: list[Tag] = field(default_factory=list)

@dataclass(frozen=True,slots=True)
class ModelSimpleRecord:
  sha256: str
  name: str
  images_count: int
  type: str = ""
  path: str = ""
  size_bytes: int = 0
  created_at: int = 0
  tags: list[Tag] = field(default_factory=list)


@dataclass(frozen=True,slots=True)
class PendingModelSimpleRecord:
  id: int
  name: str
  image: str
  tags: list[Tag] = field(default_factory=list)

File: web/model_manager_frontend/src/types/model_record.ts
export interface Prompts{
    positive: string;
    negative: string;
}

export interface MetaJson{
    description: string;
    community_links: string;
    images_count: number;
    prompts: Prompts;
}

export interface OldMetaJson{
    description: string;
    community_links: string;
    images: string[];
    prompts: Prompts;
}

export interface Tag {
  id: number;
  name: string;
}

export interface ModelRecord{
    sha256:string;
    path: string;
    name: string;
    type: string;
    size_bytes: number;
    created_at: number;
    meta_json: MetaJson;
    tags: Tag[];
}

export interface PendingModelRecord{
    id: number;
    path: string;
    sha256:string;
    name: string;
    type: string;
    size_bytes: number;
    created_at: number;
    meta_json: OldMetaJson;
    tags: Tag[];
}



export interface Model {
  sha256: string;
  name: string;
  type: string;
  path: string;
  size_bytes: number;
  created_at: number;
  tags: Tag[];
}

export interface ModelSimpleRecordWithTags{
    sha256: string;
    name: string;
    images_count: number;
    tags: Tag[];
}

export interface PendingModelSimpleRecordWithTags{
    id: number;
    name: string;
    images: string;
    tags: Tag[];
}